reference_seqs.md

```diff
@@ -0,0 +1,159 @@
+---
+permalink: /reference_seqs.html
+layout: default
+title: Htslib - Reference Sequences
+highlighting: yes
+---
+
+# Reference Sequences
+
+## Introduction
+
+CRAM typically uses reference based compression where individual bases in aligned records are compared against a known reference sequence and only bases differing are stored.
+This is used as a form of sequence compression, but it can also help construct the `MD` and `NM` auxiliary tags, further saving storing costs.
+
+Most CRAMs will use an external reference, however it is also possible to use an internal embedded reference within the CRAM file or to use no reference at all.
+
+## Search Order For Finding References
+
+When looking for an external reference the search order used by HTSlib is as follows.  
+
+1. Use any reference FASTA file specified by the `hts_set_fai_filename()` function or samtools command line options, such as `samtools view -T`.
+
+2. Search the `REF_CACHE` directory for the sequence MD5sum listed in the `@SQ` header.  This data will be unformatted raw sequence.
+
+3. In turn, search each `REF_PATH` location element for the sequence MD5sum.  This can be either local directories or remote URLs.
+
+4. Look for a local FASTA file listed in the `@SQ` header UR tag.  Earlier versions of HTSlib unintentionally also utilised http and ftp based UR tags.  These are no longer supported because they do not operate in conjunction with REF_CACHE.
+
+5. Finally if none of the above succeed, the library will return a failure code.
+
+The sections below explain these variables and formats in more detail.
+
+
+## Internal / Embedded References
+
+CRAM can store the reference internally as part of the file.
+This means the reference will not need to be known during decompression.
+While this feature was added for efficient storage of denovo assemblies, it may still be worth considering for data aligned against widely known reference genomes when wantng a self contained CRAM that has no requirements on external data.
+
+This may seem to remove the benefits of reference based encoding, but there will usually be many alignments covering the same location so this is a form of deduplication analogous to LZ compression techniques.
+The reference is stored per CRAM slice and is the portion of reference covered by the alignments within that slice.
```

Comment:
portion of reference -> portion of the reference

Maybe rewrite the entire sentence?

Reply:
I'll change it to "Each CRAM slice stores the portion of the reference covered by alignments within that slice."